time_utils.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file time_utils.cpp
+ *
+ * @brief Definitions for time utilities functions.
+ *
+ * @author Kendra Hale, <kendrahale98@gmail.com>
+ *
+ * @date 10/25/24
+ */
+#include "./time_utils.h"
+
+/**
+ * ts_is_before
+ * @brief Checks if a given timespec precedes another.
+ *
+ * @param ts1   first timespec to consider
+ * @param ts2   second timespec to consider
+ *
+ * @return true if ts1 is before ts2, false otherwise
+ */
+bool ts_is_before(const timespec& ts1, const timespec &ts2) {
+    if (ts1.tv_sec < ts2.tv_sec) {
```

Comment:
nit: 

return (ts1.tv_sec < ts2.tv_sec || ts1.tv_nsec < ts2.tv_nsec)

cpp order of operations for logical operations are such that for an or, the moment the first expression evaluates to true, it will return true

only if the first expression is false will the second evaluate.

either way, you can do this in one line.